proto/mls/api/v1/mls.proto

```diff
@@ -78,15 +78,47 @@ service MlsApi {
       body: "*"
     };
   }
+
+  // Query stored group messages
+  rpc QueryGroupMessages(QueryGroupMessagesRequest) returns (QueryGroupMessagesResponse) {
+    option (google.api.http) = {
+      post: "/mls/v1/query-group-messages"
+      body: "*"
+    };
+  }
+
+  // Query stored group messages
+  rpc QueryWelcomeMessages(QueryWelcomeMessagesRequest) returns (QueryWelcomeMessagesResponse) {
+    option (google.api.http) = {
+      post: "/mls/v1/query-welcome-messages"
+      body: "*"
+    };
+  }
+
+  // Subscribe stream of new group messages
+  rpc SubscribeGroupMessages(SubscribeGroupMessagesRequest) returns (stream xmtp.mls.message_contents.GroupMessage) {
+    option (google.api.http) = {
+      post: "/mls/v1/subscribe-group-messages"
+      body: "*"
+    };
+  }
+
+  // Subscribe stream of new welcome messages
+  rpc SubscribeWelcomeMessages(SubscribeWelcomeMessagesRequest) returns (stream xmtp.mls.message_contents.WelcomeMessage) {
+    option (google.api.http) = {
+      post: "/mls/v1/subscribe-welcome-messages"
+      body: "*"
+    };
+  }
 }
 
-// Publish a batch of MLS messages
-message PublishToGroupRequest {
-  repeated xmtp.mls.message_contents.GroupMessage messages = 1;
+// Send a batch of MLS messages
+message SendGroupMessagesRequest {
+  repeated bytes messages = 1;
```

Comment:
So, if we make some sort of breaking change to the payload format here, we are comfortable moving to a `SendGroupMessagesV2Request` and a new API? Without the wrapper proto that seems necessary.

I don't have a strong preference. We either end up having to version the payload or the endpoint.

Reply:
Hmm ok I'll define a versioned type for it so it can be consistent with the full GroupMessage versioning